File: src/Java/BehavioralPatterns/Mediator/CQRS/src/ProductsController.java
import java.math.BigDecimal;

public class ProductsController {
    private final IMediator mediator;

    public ProductsController(IMediator mediator) {
        this.mediator = mediator;
    }

    public void createProduct(String productName, BigDecimal productPrice) {
        mediator.send(new CreateProductCommand(productName, productPrice));
    }

    public Object getProducts() {
        return mediator.send(new GetProductsRequest());
    }
}

File: src/JavaScript/BehavioralPatterns/Mediator/CQRS/productsController.js

const CreateProductCommand = require("./createProductCommand");
const GetProductsRequest = require("./getProductsRequest");

class ProductsController {
    constructor(mediator) {
        this.mediator = mediator;
    }

    createProduct(productName, productPrice) {
        return this.mediator.send(new CreateProductCommand(productName, productPrice));
    }

    getProducts() {
        return this.mediator.send(new GetProductsRequest());
    }
}

module.exports = ProductsController;